loader/common/preprocessor.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+class Preprocessor(object):
+    '''
+    Runs preprocessing operations (if required)
+    '''
+    def __init__(self, newick_file, preprocessing_required=False, matching_required=False):
+        self.newick_file = newick_file
+        self.preprocessing_required = preprocessing_required
+        self.matching_required = matching_required
+
+    def trim_prefixes_and_add_root(self):
+    	with open(self.newick_file, 'r+') as newick_file:
+    		newick_string = newick_file.read().replace('\n', '')
+    	
+    		newick_string += 'root;'
```

Comment:
Wonder how this handles situations where there's already `;` at the end of the file?